### PR TITLE
fix: add missing fields to allocation query and tests to prevent future bugs

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change fixes a bug in the mapping to and from the DB for `model.Allocation`s, which caused agent labels and slots to not get persisted.

## Test Plan
- [x] add a test that covers this and makes it more difficult to miss in the future

## Commentary (optional)
eh.. 🤦 

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/db/postgres_tasks_intg_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		AgentLabel:   "something",
+		ResourcePool: "somethingelse",
+		StartTime:    time.Now().UTC().Truncate(time.Millisecond),
+	}
+	err = db.AddAllocation(aIn)
+	require.NoError(t, err, "failed to add allocation")
+
+	// Retrieve it back and make sure the mapping is exhaustive.
+	aOut, err := db.AllocationByID(aIn.AllocationID)
+	require.NoError(t, err, "failed to retreive allocation")
```
Comment (dzhu) [inline master/internal/db/postgres_tasks_intg_test.go:94]: retrieve

File: master/internal/db/postgres_tasks_intg_test.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	require.True(t, reflect.DeepEqual(aIn, aOut), pprintedExpect(aIn, aOut))
+
+	// Complete it.
+	aIn.EndTime = ptrs.TimePtr(time.Now().UTC().Truncate(time.Millisecond))
+	err = db.CompleteAllocation(aIn)
+	require.NoError(t, err, "failed to mark allocation completed")
+
+	// Re-retrieve it back and make sure the mapping is still exhaustive.
+	aOut, err = db.AllocationByID(aIn.AllocationID)
+	require.NoError(t, err, "failed to re-retreive allocation")
```
Comment (dzhu) [inline master/internal/db/postgres_tasks_intg_test.go:104]: retrieve

**Review by dzhu — APPROVED** [review]

File: master/internal/db/postgres_tasks_intg_test.go (hunk 2 above)
Comment (stoksc) [inline master/internal/db/postgres_tasks_intg_test.go:104]:
```suggestion
	require.NoError(t, err, "failed to re-retrieve allocation")
```

File: master/internal/db/postgres_tasks_intg_test.go (hunk 1 above)
Comment (stoksc) [inline master/internal/db/postgres_tasks_intg_test.go:94]:
```suggestion
	require.NoError(t, err, "failed to retrieve allocation")
```

**Pull request merged**